src/litserve/server.py

```diff
@@ -717,15 +717,16 @@ def run(
         elif uvicorn_worker_type is None:
             uvicorn_worker_type = "process"
 
-        servers = self._start_server(port, num_uvicorn_servers, log_level, sockets, uvicorn_worker_type)
-
-        for s in servers:
-            s.join()
-
-        for w in litserve_workers:
-            w.terminate()
-            w.join()
-        manager.shutdown()
+        try:
+            servers = self._start_server(port, num_uvicorn_servers, log_level, sockets, uvicorn_worker_type)
+            for s in servers:
+                s.join()
+        finally:
+            print("Shutting down LitServe")
+            for w in litserve_workers:
+                w.terminate()
+                w.join()
+            manager.shutdown()
```

Comment:
why the finally is needed?

Reply:
so that we always make sure that the LitServer workers are terminated after terminating the uvicorn server

Reply:
so in fact, you do?
```py
try:
    servers = self._start_server(port, num_uvicorn_servers, log_level, sockets, uvicorn_worker_type)
    for s in servers:
        s.join()
    print("Shutting down LitServe")
    for w in litserve_workers:
        w.terminate()
        w.join()
    manager.shutdown()
except:
    pass
```